apps/web/src/components/Composer/Actions/SpaceSettings/index.tsx

```diff
@@ -0,0 +1,187 @@
+import MenuTransition from '@components/Shared/MenuTransition';
+import { Menu } from '@headlessui/react';
+import { CheckIcon, ChevronDownIcon } from '@heroicons/react/outline';
+import { Input, Toggle } from '@lenster/ui';
+import { Trans } from '@lingui/macro';
+import clsx from 'clsx';
+import type { FC, ReactNode } from 'react';
+import React, { useState } from 'react';
+
+interface SpaceSettingsProps {
+  children: ReactNode;
+}
+
+enum tokenGateCondition {
+  HAVE_A_LENS_PROFILE = 'have a lens profile',
+  FOLLOW_A_LENS_PROFILE = 'follow a lens profile',
+  COLLECT_A_POST = 'collect a post',
+  MIRROR_A_POST = 'mirror a post'
```

Comment:
Suggest to have something like this `HAVE_A_LENS_PROFILE = HAVE_A_LENS_PROFILE` and add condition on component

Because we need to wrap text under i18n component, if its inside enum its not possible 🙇🏼

Reply:
Updated